### PR TITLE
Fix enum pattern matching

### PR DESCRIPTION
**Pull request by GerardSmit**
When using `is` pattern matching with enums, the enum value is dropped. For example:

```cs
_ = System.Data.ConnectionState.Closed is System.Data.ConnectionState.Closed;
```

Gets converted to:

```cs
_ = System.Data.ConnectionState.Closed is global::System.Data.ConnectionState;
```

This PR checks if the type is an enum and doesn't match the base `typeSyntax` (because the right-side is missing) and then adds it back.

### CODE REVIEW

File: src/Zomp.SyncMethodGenerator/AsyncToSyncRewriter.cs [hunk 1]
```diff
@@ -2088,7 +2088,14 @@ private bool RemoveDeclarator(VariableDeclaratorSyntax variable)
     private TypeSyntax ProcessSyntaxUsingSymbol(TypeSyntax typeSyntax)
     {
         var typeSymbol = semanticModel.GetTypeInfo(typeSyntax).Type;
-        return typeSymbol is null ? typeSyntax : ProcessSymbol(typeSymbol).WithTriviaFrom(typeSyntax);
+
+        return typeSymbol switch
+        {
+            null => typeSyntax,
+            { TypeKind: TypeKind.Enum } when typeSyntax is QualifiedNameSyntax { Right: IdentifierNameSyntax r } && (r.Identifier.ValueText != typeSymbol.Name || typeSymbol.ToString() != typeSyntax.WithoutTrivia().ToString())
```
Comment (GerardSmit) [inline src/Zomp.SyncMethodGenerator/AsyncToSyncRewriter.cs:2095]: I've added a quick check with `r.Identifier.ValueText != typeSymbol.Name` which is almost always true, because `typeSymbol.ToString()` is allocating a string. The edge case is if the enum value and enum have the same name, e.g.

```cs
enum Test { Test }
```

This would fail with `r.Identifier.ValueText != typeSymbol.Name`. I've added a unit test for this.

File: tests/Generator.Tests/Snapshots/TypeTests.EnumPattern#g.verified.cs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+﻿//HintName: Test.Class.MethodAsync.g.cs
+_ = System.Data.ConnectionState.Closed is global::System.Data.ConnectionState.Closed;
```
Comment (virzak) [inline tests/Generator.Tests/Snapshots/TypeTests.EnumPattern#g.verified.cs:2]: I think it should be:

```
_ = global::System.Data.ConnectionState.Closed is global::System.Data.ConnectionState.Closed;
```

**Review by virzak — APPROVED** [review]

**Comment by virzak** [comment]
Thanks a lot for the PR.

I expanded your test case to cover using static class.

The change is in the latest nuget package.
